mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/intercepting/LoadingInvocationContext.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.intercepting;
+
+import java.util.Map;
+
+/**
+ * Exposes contextual information about the intercepted invocation and
+ * operations that enable interceptor methods to control the behavior
+ * of the invocation chain.
+ * @param <O> The options for loading process
+ */
+public interface LoadingInvocationContext<O> {
+
+	/**
+	 * @return The mass options for loading process.
+	 */
+	O options();
+
+	/**
+	 * @return the identifier of the tenant.
+	 */
+	String tenantId();
+
+	/**
+	 * Evaluates this predicate on the intercepting process.
+	 *
+	 * @param active a predicate that will be test intercepting active process
+	 */
+	void active(LoadingProcessActivePredicate active);
+
+	/**
+	 * Return this predicate on the intercepting process.
+	 *
+	 * @return a predicate that will be test intercepting active process
+	 */
+	LoadingProcessActivePredicate active();
```

Comment:
It really isn't just about transactions. The idea is to isolate interception from entity loading. The flag allows you to break loops from loading entities from the interception level, no matter what the cause in the interception. The transaction is just one example. I'll try to write a use case test.

Reply:
All in all, this can be replaced by injecting your own context into interactions and recalling it in loading:
```java
			@Override
			public List<?> next() {
				int batchSize = context.options().batchSize();

				List<Object> destination = new ArrayList<>( batchSize );
				while ( iterator.hasNext() ) {
					destination.add( iterator.next() );
					if ( !context.context( LoadingThreadOptions.class ).interrupted() ) {
						throw log.contextInterruptedWhileProducingIdsForBatchIndexing( context.includedEntityNames() );
					}

					if ( destination.size() == batchSize ) {
						return destination;
					}
				}
				return destination;
			}
}
```
I will remove all functionality of active().